src/BladeFactory.php

```diff
@@ -65,7 +87,15 @@ public static function getFacadeAccessor()
         $config->set('view.compiled', $pathToCompiledTemplates);
         $container['config'] = $config;
 
-        $bladeCompiler->component('dynamic-component', DynamicComponent::class);
+        $bladeCompiler->component(DynamicComponent::class, 'dynamic-component');
```

Comment:
Yes, because that’s the way how the blade compiler reads the parameters (BladeCompiler.php:L633 `public function component($class, $alias = null, $prefix = '')`. If the paremeters are given in wrong order (as before), that’s usually not a problem unless the component class is defined in the global namespace and thus does not contain a backslash, which is used to detect which parameter is the class and which one the alias.